src/app/shared/services/auth.service.ts

```diff
@@ -46,11 +46,32 @@ export class AuthService {
 
   login(user: User): Observable<boolean> {
     const users = this.users.getValue();
-    const item = !!users.find((i) => (i.email === user.name || i.name === user.name) && i.password === user.password);
-    if (!item) {
+    const authUser = users.find((i) => (i.email === user.name || i.name === user.name) && i.password === user.password);
+    if (authUser === undefined) {
       return of(false);
     }
+
+    this.authorizedUser.next(authUser);
     this.authorized.next(true);
     return of(true);
   }
+
+  logout(): Observable<boolean> {
+    this.authorized.next(false);
+    return this.authorized.asObservable();
+  }
+
+  delete(): Observable<boolean> {
```

Comment:
please add id to param and use for search

Reply:
Done

Reply:
just id, not whole user object

Reply:
Done

Reply:
not done

Reply:
Your realization will delete all users with the same name